### PR TITLE
Add cloneEngine function to random number service

### PR DESCRIPTION
**Pull request by wddgit**
#### PR description:

Add cloneEngine function to the random number service. This is intended to support the work Chris is doing on external processes which is related to generators. @Dr15Jones 

#### PR validation:

Extended an existing unit test to verify the new function works.

### CODE REVIEW

**Comment by wddgit** [comment]
please test

File: FWCore/Utilities/interface/RandomNumberGenerator.h [hunk 1]
```diff
@@ -157,6 +158,8 @@ namespace edm {
     /// Use this engine in the global begin luminosity block method
     virtual CLHEP::HepRandomEngine& getEngine(LuminosityBlockIndex const&) = 0;
 
+    virtual std::unique_ptr<CLHEP::HepRandomEngine> cloneEngine(LuminosityBlockIndex const&) = 0;
```
Comment (makortel) [inline FWCore/Utilities/interface/RandomNumberGenerator.h]: Could you add a short comment explaining the intended use case (including something around "not intended for general use")?

Comment (wddgit) [reply FWCore/Utilities/interface/RandomNumberGenerator.h]: Done. Good  idea.

**Comment by wddgit** [comment]
please test

**Comment by makortel** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**